src/EnergyPlus/UnitVentilator.cc

```diff
@@ -439,193 +351,147 @@ namespace UnitVentilator {
-                                      NodeInputManager::CompFluidStream::Primary,
-                                      ObjectIsNotParent);
+            if (!unitVent.ATMixerExists) {
+                unitVent.AirInNode = NodeInputManager::GetOnlySingleNode(state,
+                                                                         Alphas(6),
+                                                                         ErrorsFound,
+                                                                         DataLoopNode::ConnectionObjectType::ZoneHVACUnitVentilator,
+                                                                         Alphas(1) + "-OA MIXER",
+                                                                         DataLoopNode::NodeFluidType::Air,
+                                                                         DataLoopNode::ConnectionType::Inlet,
+                                                                         NodeInputManager::CompFluidStream::Primary,
+                                                                         DataLoopNode::ObjectIsNotParent);
             }
 
-            state.dataUnitVentilators->UnitVent(UnitVentNum).FanType = Alphas(11);
-            state.dataUnitVentilators->UnitVent(UnitVentNum).FanName = Alphas(12);
+            unitVent.FanType = Alphas(11);
+            unitVent.FanName = Alphas(12);
             errFlag = false;
-            ValidateComponent(state,
-                              state.dataUnitVentilators->UnitVent(UnitVentNum).FanType,
-                              state.dataUnitVentilators->UnitVent(UnitVentNum).FanName,
-                              errFlag,
-                              CurrentModuleObject);
+            ValidateComponent(state, unitVent.FanType, unitVent.FanName, errFlag, CurrentModuleObject);
             if (errFlag) {
-                ShowContinueError(state,
-                                  "specified in " + CurrentModuleObject + " = \"" + state.dataUnitVentilators->UnitVent(UnitVentNum).Name + "\".");
+                ShowContinueError(state, "specified in " + CurrentModuleObject + " = \"" + unitVent.Name + "\".");
                 ErrorsFound = true;
             } else {
-                if (!UtilityRoutines::SameString(state.dataUnitVentilators->UnitVent(UnitVentNum).FanType, "Fan:SystemModel")) {
-                    GetFanType(state,
-                               state.dataUnitVentilators->UnitVent(UnitVentNum).FanName,
-                               state.dataUnitVentilators->UnitVent(UnitVentNum).FanType_Num,
-                               errFlag,
-                               CurrentModuleObject,
-                               state.dataUnitVentilators->UnitVent(UnitVentNum).Name);
+                if (!UtilityRoutines::SameString(unitVent.FanType, "Fan:SystemModel")) {
+                    Fans::GetFanType(state, unitVent.FanName, unitVent.FanType_Num, errFlag, CurrentModuleObject, unitVent.Name);
```

Comment:
Is this a more complicated thing than `getEnumerationValue`? Why so many arguments?